pkg/instancetype/preference/apply/device_test.go

```diff
@@ -248,4 +246,30 @@ var _ = Describe("Preference.Devices", func() {
 			}
 		})
 	})
+
+	Context("PreferredTPM", func() {
+		DescribeTable("should",
+			func(vmiTPM, preferenceTPM, expectedTPM *virtv1.TPMDevice) {
+				vmi.Spec.Domain.Devices.TPM = vmiTPM
+				preferenceSpec.Devices.PreferredTPM = preferenceTPM
+				Expect(vmiApplier.ApplyToVMI(field, instancetypeSpec, preferenceSpec, &vmi.Spec, &vmi.ObjectMeta)).To(Succeed())
+				Expect(vmi.Spec.Domain.Devices.TPM).To(Equal(expectedTPM))
+			},
+			Entry("only apply when TPM device is nil within VMI spec",
+				nil,
+				&virtv1.TPMDevice{Persistent: pointer.P(true)},
+				&virtv1.TPMDevice{Persistent: pointer.P(true)},
+			),
+			Entry("not apply when TPM device is provided within VMI spec",
+				&virtv1.TPMDevice{Persistent: pointer.P(true)},
+				&virtv1.TPMDevice{},
+				&virtv1.TPMDevice{Persistent: pointer.P(true)},
+			),
+			Entry("not apply when TPM device is provided and disabled within VMI spec",
+				&virtv1.TPMDevice{Enabled: pointer.P(false)},
+				&virtv1.TPMDevice{},
```

Comment:
Set it to true here?

Reply:
`Enabled` defaults the `true` but I can add an explicit test case.